airbyte-ci/connectors/metadata_service/lib/metadata_service/spec_cache.py

```diff
@@ -56,3 +61,9 @@ def list_cached_specs() -> List[CachedSpec]:
     blobs = bucket.list_blobs(prefix=CACHE_FOLDER)
 
     return [get_docker_info_from_spec_cache_path(blob.name) for blob in blobs]
+
+
+def get_cached_spec(spec_cache_path: str) -> dict:
+    client = storage.Client.create_anonymous_client()
+    bucket = client.bucket(SPEC_CACHE_BUCKET_NAME)
+    return json.loads(bucket.blob(spec_cache_path).download_as_string())
```

Comment:
@bnchrch should we download specs from the metadata_service lib or in the Dagster realm?

Reply:
Honestly, Im neutral on this one.

To me gcs blobs are used enough to be considered a "type" that we could build a contract between two systems on.

So if the metadata service returns a list of gcs blobs for the metadata files and its up to the calling service if they download them or something else. Thats fine in my books. 

But its also fine if the service returns the raw text from the file.

Reply:
Ok, let's keep it there. We might have non-orchestrator use cases that require downloading a spec.